### ISSUE TITLE
Issue with remaking a variable rate jump problem

### ISSUE THREAD

**Issue by calebrich** [SciML/JumpProcesses.jl#446]
When I try to remake a JumpProblem which has a VariableRateJump, I get an error. For example, with the following code:

```
using DifferentialEquations

function f(du, u, p, t)
    du[1] = u[1]
    nothing
end

prob = ODEProblem(f, [0.2], (0.0, 10.0))
rate(u, p, t) = 2

function affect!(integrator)
    integrator.u[1] = integrator.u[1] / 2
    nothing
end

var_jump = VariableRateJump(rate, affect!)

jump_prob = JumpProblem(prob, Direct(), var_jump)

jump_prob2 = remake(jump_prob, u0 = [0.3])

sol = solve(jump_prob2, Tsit5())
```

I get the error
`AssertionError: jump_prob.prob.u0 isa ExtendedJumpArray`

**Comment by isaacsas** [comment]
Thanks for reporting! I’ll try to look into this as soon as I have a chance.

**Comment by isaacsas** [comment]
Should hopefully be fixed in 9.13.4 but please feel free to open another issue if there are still problems.

**Comment by calebrich** [comment]
Yes, that works. Thanks for the speedy fix!
